extension-compose/src/main/java/com/mapbox/maps/extension/compose/internal/MapApplier.kt

```diff
@@ -0,0 +1,83 @@
+package com.mapbox.maps.extension.compose.internal
+
+import androidx.compose.runtime.AbstractApplier
+import com.mapbox.maps.MapView
+
+/**
+ * Defines the contract of a MapNode, the MapNode will be notified when it's added/moved/removed from
+ * the node tree.
+ */
+internal interface MapNode {
+  /**
+   * Invoked when the [MapNode] is attached to the MapboxMap node tree.
+   */
+  fun onAttached() {}
+
+  /**
+   * Invoked when the [MapNode] is moved inside the node tree.
+   *
+   * @param from original position
+   * @param to the target position
+   */
+  fun onMoved(from: Int, to: Int) {}
+
+  /**
+   * Invoked when the [MapNode] is removed from the MapboxMap node tree.
+   */
+  fun onRemoved() {}
```

Comment:
```suggestion
  fun onDetached() {}
```
because we have `attached`

Reply:
it's following the same name pattern as the abstract class, node can be removed from the node tree, and it's not guaranteed it could be reattached to the node tree.